### ISSUE TITLE
Validation dataset

### ISSUE THREAD

**Issue by alainrichardt** [maxhodak/keras-molecules#23]
Im curious - can anybody help me understand why there's a separate validation dataset for the auto-encoder?

If the purpose of the autoencoder is to take us from (categorical, discreet) to (multinomial, continuous) and then back to (categorical discreet) then isn't it better if the network is trained on all of the samples, even to the point of overfitting? I'm saying this because we're not trying to train the network to generalize molecular structure - it's simply a transformer. Once we construct the latent space then it's up to us to analyse it, but shouldn't the autoencoder simply be trained to encode and decode, and not to generalize? 

We still get predictive ability by manipulating transforms through the latent space (as a totally separate task), then the decoder can spit out the molecule that was the result of our latent transform.

No?

**Comment by dribnet** [comment]
I don't think you want want your network to overfit (eg: memorize the training set) because eventually you want to use the model to encode molecules not seen in the training set, right? A validation dataset provides metrics on how well the model is doing for a single run, and across multiple runs can be used to tune hyperparameters like `--latent_dim` to an optimal value. Also, keep in mind a latent space can support all sorts of other operations, such as classification, interpolation, extrapolation, analogy, attribute vectors, etc. But to test these operations, it probably best to use data not in the training set.

**Comment by pechersky** [comment]
I agree with you, @alainrichardt, that it is not totally necessary to have a separate validation set. @dribnet makes good points on how you might utilize a validation set. However, I want to point out that the VAE doesn't promise to give a consistent latent space representation for a particular input, nor a particular decoding of a latent space point, so "accuracy" is a fuzzy measure that depends on exactly how your input is autoencoded at that instance. More than just wanting to encode molecules outside of the training set, you would also like to decode sampled latent space points that aren't representations of your dataset, and verify that calculated characteristics are valid. Then, your "validation" set might not actually be molecules, but sampled regions near known points in the latent space.
